test/most-w3msg-test.js

```diff
@@ -1,88 +1,92 @@
-		}, 0);
-
-		return s.take(1).observe(spy).then(function() {
-			expect(spy).toHaveBeenCalledOnceWith(sentinel);
-		});
-	});
-
-	it('should call disposer when stream ends', function() {
-		var es = new fakes.FakeEventSource();
-
-		var spy = this.spy();
-		var s = mws.fromEventSource(es, spy);
-
-		setTimeout(function() {
-			es.emit('message', sentinel);
-		}, 0);
-
-		return s.take(1).drain().then(function() {
-			expect(spy).toHaveBeenCalledOnce();
-		});
-	});
-});
-
-describe('fromEventSourceOn', function() {
-	it('should contain events received by EventSource', function() {
-		var es = new fakes.FakeEventSource();
-
-		var s = mws.fromEventSourceOn('test', es, es.close.bind(es));
-		var spy = this.spy();
-
-		setTimeout(function() {
-			es.emit('test', sentinel);
-		}, 0);
-
-		return s.take(1).observe(spy).then(function() {
-			expect(spy).toHaveBeenCalledOnceWith(sentinel);
-		});
-	});
-});
\ No newline at end of file
+/** @license MIT License (c) copyright 2016 original author or authors */
+
+import {describe, it} from 'mocha'
+import assert from 'assert'
+import sinon from 'sinon'
+
+import * as fakes from './helper/fakes'
+import * as mostW3msg from '../src/most-w3msg'
+
+const sentinel = { value: 'sentinel' }
+
+describe('most-w3msg', () => {
+  describe('fromWebSocket', () => {
+    it('should contain messages received by WebSocket', done => {
+      const observer = sinon.spy()
+      const ws = new fakes.FakeWebSocket()
+      const s = mostW3msg.fromWebSocket(ws, ws.close.bind(ws))
+
+      setTimeout(() => ws.send(sentinel), 0)
+
+      return s.take(1).observe(observer).then(() => {
+        assert.strictEqual(observer.callCount, 1)
+        assert(observer.calledWith(sentinel))
+        done()
+      })
+    })
+
+    it('should call disposer when stream ends', done => {
+      const observer = sinon.spy()
+      const ws = new fakes.FakeWebSocket()
+      const s = mostW3msg.fromWebSocket(ws, observer)
+
+      setTimeout(() => {
+        ws.send(sentinel)
+      }, 0)
+
+      return s.take(1).drain().then(() => {
+        assert.strictEqual(observer.callCount, 1)
+        done()
+      })
+    })
+  })
+
+  describe('fromEventSource', () => {
+    it('should contain messages received by EventSource', done => {
+      const observer = sinon.spy()
+      const es = new fakes.FakeEventSource()
+      const s = mostW3msg.fromEventSource(es, es.close.bind(es))
+
+      setTimeout(() => {
+        es.emit('message', sentinel)
+      }, 0)
+
+      return s.take(1).observe(observer).then(() => {
+        assert.strictEqual(observer.callCount, 1)
+        assert(observer.calledWith(sentinel))
+        done()
+      })
+    })
+
+    it('should call disposer when stream ends', done => {
+      const observer = sinon.spy()
+      const es = new fakes.FakeEventSource()
+      const s = mostW3msg.fromEventSource(es, observer)
+
+      setTimeout(() => {
+        es.emit('message', sentinel)
+      }, 0)
```

Comment:
The reason `jsinspect` is complaining is the similarity between lines 62-68 and 46-52.  It looks like the only difference is `es.close.bind(es)` vs `observer` in the call to `fromEventSource`, so you could probably extract a helper function.